File: src/main/java/net/ccbluex/liquidbounce/features/module/modules/render/HUD.java
/*
 * LiquidBounce Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge.
 * https://github.com/CCBlueX/LiquidBounce/
 */
package net.ccbluex.liquidbounce.features.module.modules.render;

import lol.liquidcat.event.*;
import lol.liquidcat.LiquidCat;
import lol.liquidcat.features.module.Module;
import lol.liquidcat.features.module.ModuleCategory;
import lol.liquidcat.features.module.ModuleInfo;
import net.ccbluex.liquidbounce.ui.client.hud.designer.GuiHudDesigner;
import lol.liquidcat.value.BoolValue;
import net.minecraft.client.gui.GuiChat;
import net.minecraft.util.ResourceLocation;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;

@ModuleInfo(name = "HUD", description = "Toggles visibility of the HUD.", category = ModuleCategory.RENDER, array = false)
@SideOnly(Side.CLIENT)
public class HUD extends Module {
    public final BoolValue blackHotbarValue = new BoolValue("BlackHotbar", true);
    public final BoolValue inventoryParticle = new BoolValue("InventoryParticle", false);
    private final BoolValue blurValue = new BoolValue("Blur", false);
    public final BoolValue fontChatValue = new BoolValue("FontChat", false);

    public HUD() {
        setState(true);
    }

    @EventTarget
    public void onRender2D(final Render2DEvent event) {
        if (mc.currentScreen instanceof GuiHudDesigner)
            return;

        LiquidCat.hud.render(false);
    }

    @EventTarget
    public void onUpdate(final UpdateEvent event) {
        LiquidCat.hud.update();
    }

    @EventTarget
    public void onKey(final KeyEvent event) {
        LiquidCat.hud.handleKey('a', event.getKey());
    }

    @EventTarget(ignoreCondition = true)
    public void onScreen(final ScreenEvent event) {
        if (mc.theWorld == null || mc.thePlayer == null)
            return;

        if (getState() && blurValue.get() && !mc.entityRenderer.isShaderActive() && event.getGuiScreen() != null &&
                !(event.getGuiScreen() instanceof GuiChat || event.getGuiScreen() instanceof GuiHudDesigner))
            mc.entityRenderer.loadShader(new ResourceLocation(LiquidCat.CLIENT_NAME.toLowerCase() + "/blur.json"));
        else if (mc.entityRenderer.getShaderGroup() != null &&
                mc.entityRenderer.getShaderGroup().getShaderGroupName().contains("liquidbounce/blur.json"))
            mc.entityRenderer.stopUseShader();
    }
}

File: src/main/kotlin/lol/liquidcat/features/module/modules/render/HUD.kt
/*
 * LiquidCat Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge.
 * https://github.com/CatsDevelopment/LiquidCat
 */
package lol.liquidcat.features.module.modules.render

import lol.liquidcat.LiquidCat
import lol.liquidcat.event.EventTarget
import lol.liquidcat.event.KeyEvent
import lol.liquidcat.event.Render2DEvent
import lol.liquidcat.event.UpdateEvent
import lol.liquidcat.features.module.Module
import lol.liquidcat.features.module.ModuleCategory
import lol.liquidcat.ui.client.hud.designer.GuiHudDesigner
import lol.liquidcat.utils.render.GLUtils
import lol.liquidcat.value.BoolValue
import net.minecraft.client.gui.GuiChat

object HUD : Module("HUD", "Toggles visibility of the HUD.", ModuleCategory.RENDER, hide = true) {

    val blackHotbar by BoolValue("BlackHotbar", true)
    private val blur by BoolValue("Blur", false)
    val fontChat by BoolValue("FontChat", false)
    val cleanChat by BoolValue("CleanChat", true)

    @EventTarget
    fun onRender2D(event: Render2DEvent) {
        if (blur && mc.currentScreen != null && !(mc.currentScreen is GuiChat || mc.currentScreen is GuiHudDesigner))
            GLUtils.blur(10) {
                GLUtils.drawRect(0f, 0f, mc.displayWidth.toFloat(), mc.displayHeight.toFloat(), -1)
            }

        if (mc.currentScreen !is GuiHudDesigner)
            LiquidCat.hud.render(false)
    }

    @EventTarget
    fun onUpdate(event: UpdateEvent) {
        LiquidCat.hud.update()
    }

    @EventTarget
    fun onKey(event: KeyEvent) {
        LiquidCat.hud.handleKey('a', event.key)
    }

    init {
        state = true
    }
}